In [40]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly
import plotly.plotly as py
import plotly.figure_factory as ff
import warnings

sns.set(style="whitegrid")

In [41]:
# suppress warnings

warnings.filterwarnings('ignore')

<style>
//change position of header 3 {}
h3 {
    position: relative;
}

//add class for 2 columns {}
div.column2 {
    float: left;
    width: 50%;
}

//add class for 3 columns {}
div.column3 {
    float: left;
    width: 33.33%;
}

/* Clear floats after the columns */
div.row:after {
    content: "";
    display: table;
    clear: both;
}

div.footer {
    position: fixed;
    left: 0;
    bottom: 45px;
    width: 100%;
    background-color: transparent;
    color: black;
    text-align: center;
}

//change background settings {}
div.slide-background {
	border-top: 30px #660000 solid;
	border-bottom: 30px #660000 solid;
}

//class for margin top -60 px {}
.big-img {
    margin-top: -60px;
}

//class for margin top -30 px {}
.med-img {
    margin-top: -30px;
}

* {
    transition: none;
}

</style>

# Predicting the Mortality Rate Due to Opioid Addiction At the County Level

### Daniel Lee
#### August 17, 2018

<div class='footer'>Slides created using Jupyter Notebook, RISE, and reveal.js</div>

<section data-background-transition='none'>

Test

</section>

<div class="column3" style="position: fixed; top: 15vh; left: 40vw;">
    <ul>
        <li>Health Problems</li>
        <img src="images/health_problems.jpg">
    </ul>
</div>
    

  
  
  <div class="row">

  <div class="column3"><ul><li>Financial Problems</ul><img src="images/financial_problems.jpg"></div>
  

  <div class="row">

  <div class="column3"><ul><li>Relational Problems</ul><img src="images/relational_problems.jpg"></div>
</div>

## What is the opioid epidemic?

## Annual Cause of Death by Opioid Overdose

![alt text](images/annual_deaths_caused_by_opioid_overdose.png)

### Number of Deaths by Demographic Factors 1999 - 2016

![](images/who_are_the_victims.png)

### Living with an Opioid Addiction  

<iframe style="margin-top:20px; margin-left: auto; margin-right: auto" width="420" height="315" src="https://www.youtube.com/embed/VpEQaa6jzUU">
</iframe>

### Living with an Opioid Addiction  

[![Alt text](https://img.youtube.com/vi/VpEQaa6jzUU/0.jpg)](https://www.youtube.com/watch?v=VpEQaa6jzUU)


### Reading Between the Numbers - Why We Should Care

Many are dying because of opioid overdose. But it's not just death. Many people that are not represented in these death numbers are struggling with opioid addiction. Not only does opioid addiction destroy the health of the addicted individual, but it also negatively impacts the lives of the addicted person's loved ones, his or her families, and, ultimately, the communities at large. Dangers of opioid addiction to communities have been recognized by other nations as well. For example, in the 1950s, because of the destroyed lives and communities by opioid addiction, China not only banned opium, but it also executed those who were caught selling opium. 

Addiction to opioids, like any other addictions, leads to financial problems, health problems, lack of self control, relational difficulties, and ultimately destroys every aspect of the person, and the end result is death. Watch the video (https://youtu.be/VpEQaa6jzUU) of the story of Amy and her struggle with opioid addiction to see a glimpse of the struggles that come with opioid addiction.

* Not represented in the death numbers are those who are struggling with opioid addiction

* Addiction to opioids, like any other addictions, leads to 

<div class="row">
    <div class="column3"><ul><li>Health Problems</ul><img src="images/health_problems.jpg"></div>
  <div class="column3"><ul><li>Financial Problems</ul><img src="images/financial_problems.jpg"></div>
  <div class="column3"><ul><li>Relational Problems</ul><img src="images/relational_problems.jpg"></div>
</div>

# Opioids

* Diverse class of moderately strong painkillers

<div class="row">
  <div class="column3"><ul><li>Oxycodone</ul><img src="images/oxycodone.jpg"></div>
  <div class="column3"><ul><li>Hydrocodone</ul><img src="images/vicodin.jpg"></div>
  <div class="column3"><ul><li>Fentanyl</ul><img src="images/fentanyl.jpg"></div>
</div>

# Problem

<ul style="padding-left:40px">
    <li>Can we predict how bad the problem will be in a county?</li>
</ul>

<ul style="padding-left:80px">
    <li>Useful to help allocate resources</li>
</ul>

<ul style="padding-left:80px">
    <li>Learn more about causes of opioid overdose</li>
</ul>

<ul style="padding-left:80px">
    <li>Make biggest impact</li>
</ul>

# Problem

<ul style="padding-left:40px">
    <li>Predict the county's mortality rate caused by drug overdose for 2016 given various independent variables</li>
</ul>

<ul style="padding-left:80px">
    <li>County's median household income</li>
</ul>

<ul style="padding-left:80px">
    <li>Population estimates</li>
</ul>

<ul style="padding-left:80px">
    <li>Unemployment</li>
</ul>

<ul style="padding-left:80px">
    <li>Poverty rate estimates</li>
</ul>

<ul style="padding-left:80px">
    <li>Educational attainment</li>
</ul>

<ul style="padding-left:80px">
    <li>Opioid prescription rate</li>
</ul>

## USA by Opioid Overdose Mortality Rate, 2016

![](images/newplot.png)

### Top Ten Counties with Highest Mortality Rate Caused by Opioid Overdose in 2016 (using class)
<div class="big-img">
<img src="images/top_ten_counties_highest_opioid_mortality.png" />
</div>

### Top Ten Counties with Highest Mortality Rate Caused by Opioid Overdose in 2016

<img style="margin-top:-60px;" src="images/top_ten_counties_highest_opioid_mortality.png" />

### County with Highest Mortality Rate Due to Opioid Addiction

<ul style="padding-left:40px">
    <li>Highest county: <b> Harrison, KY</b></li>
</ul>

<ul style="padding-left:80px">
    <li>118 deaths per 100,000 people due to opioid overdose</li>
</ul>

<div class="row">
  <div class="column2"><ul><img src="images/kentucky.png"></div>
  <div class="column2"><ul><img src="images/harrison_ky.png"></div>
</div>

### Description of Data

* US County Opioid Prescribing Rates 2016
    * https://www.cdc.gov/drugoverdose/maps/rxcounty2016.html
* Unemployment, median household income, educational attainment, poverty rate, and population:
    * https://www.ers.usda.gov/data-products/county-level-data-sets/download-data/
* Underlying Cause of Death
    * https://wonder.cdc.gov/controller/saved/D77/D39F042

## Crude Opioid Mortality Rate

<ul style="padding-left:40px">
    <li>Estimated rate for deaths caused by opioid overdose in the county for 2016 (per 100,000 people)</li>
</ul>

<ul style="padding-left:80px">
    <li>Specifically, the types of drug-related deaths include the following:</li>
</ul>

<ul style="padding-left:80px">
    <li>Drug poisonings (overdose)</li>
</ul>

<ul style="padding-left:80px">
    <li>Unintentional</li>
</ul>

<ul style="padding-left:80px">
    <li>Suicide</li>
</ul>

<ul style="padding-left:80px">
    <li>Homicide</li>
</ul>

<ul style="padding-left:80px">
    <li>Undetermined</li>
</ul>

## Crude Opioid Mortality Rate

<ul style="padding-left:40px">
    <li>The specific drugs included in the death rates are the following:</li>
</ul>

<ul style="padding-left:80px">
    <li>Opium</li>
</ul>

<ul style="padding-left:80px">
    <li>Heroin</li>
</ul>

<ul style="padding-left:80px">
    <li>Opther opioids</li>
</ul>

<ul style="padding-left:80px">
    <li>Methadone</li>
</ul>

<ul style="padding-left:80px">
    <li>Other synthetic narcotics</li>
</ul>

<ul style="padding-left:80px">
    <li>Other unspecified narcotics</li>
</ul>

### Distribution of County Level Opioid Overdose Mortality Rate

![](images/violinplot_only.png)

### Distribution of County Level Opioid Overdose Mortality Rate

![](images/violinplot_histogram.png)

### Distribution of County Level Opioid Overdose Mortality Rate

![](images/violinplot_histogram_cdf.png)

### Distribution of County Level Opioid Overdose Mortality Rate

![](images/violinplot_histogram_cdf_one_annotation.png)

### Distribution of County Level Opioid Overdose Mortality Rate

![](images/distribution_of_opioid_overdose_mortality_rate.png)

## Violin Plots of All Variables Except Crude Mortality Rate
<img style="margin-top:-16px;" src="images/violin_plots.png" />

### Correlation Heat Map of All the Variables

<img style="margin-top:-26px;" src="images/correlation_heat_map.png" />

#### Pearson's Correlation Between Crude Mortality Rate and All Other Variables

<img style="margin-top:-17px;" src="images/correlation_crude_mortality_rate_all_other_variables.png" />

### Scatter Plots of Crude Opioid Mortality Rate and Other Variables
<img style="margin-top:-40px;" src="images/scatter_plots_crude_opioid_mortality_other_variables_reshaped.png" />

#### Hypothesis Test of Correlation By Permuting Samples - Distribution of Correlation Values
<img style="margin-top:-17px;" src="images/hyp_test_correlation.png" />

### Linear Regression and Confidence Interval Using Pairs Bootstrap
<img style="margin-top:-20px;" src="images/bootstrap_linear_regression.png" />

### Conclusion From Exploratory Data Analysis

<ul style="padding-left:40px">
    <li>Counties with higher opioid addiction mortality rates have</li>
</ul>

<ul style="padding-left:80px">
    <li>Higher GQ estimates</li>
</ul>

<ul style="padding-left:80px">
    <li>Higher population estimates</li>
</ul>

<ul style="padding-left:80px">
    <li>Higher median household income</li>
</ul>

### Conclusion From Exploratory Data Analysis

<ul style="padding-left:40px">
    <li>Further research</li>
</ul>

<ul style="padding-left:80px">
    <li>Why do counties that have higher population living in group quarters have higher opioid overdose mortality rates?</li>
</ul>

<ul style="padding-left:120px">
    <li>Possibly consider looking into specific group quarters estimates by county</li>
</ul>

# In Depth Analysis

## Procedures

<ul style="padding-left:40px">
    <li>For a given regression prediction algorithm, do the following:</li>
</ul>

<ul style="padding-left:80px">
    <li>Split data into 80% train set and 20% test set</li>
</ul>

<ul style="padding-left:80px">
    <li>Use default hyperparameter values to fit data (no cross-validation used)</li>
</ul>

<ul style="padding-left:80px">
    <li>Calculate RMSE</li>
</ul>

<ul style="padding-left:80px">
    <li>Generate Bootstrap Sampling Distribution of RMSE (10000 bootstrap samples)</li>
</ul>

<ul style="padding-left:80px">
    <li>Tune hyperparameter values using RandomizedSearchCV (1000 iterations) and 5-fold cross-valudation</li>
</ul>

<ul style="padding-left:80px">
    <li>Select model with lowest RMSE</li>
</ul>

<ul style="padding-left:80px">
    <li>Generate Bootstrap Sampling Distribution of RMSE (10000 bootstrap samples)</li>
</ul>

# Bootstrap RMSE Distributions for All Methods

<img src="images/bootstrap_hist_1.png" />

# Bootstrap RMSE Distributions for All Methods

<img src="images/bootstrap_hist_2.png" />

## Boxplot of Bootstrap RMSE Distributions for All Methods
<img src="images/boxplot_bootstrap_dist.png" />

## Predicted vs Actual Opioid Mortality Rate
<img style="margin-top:-15px;" src="images/predicted_actual_regression_all.png" />

## Predicted vs Actual Opioid Mortality Rate
<img style="margin-top:-15px;" src="images/predicted_actual_regression.png" />

### Feature Importances
<img style="margin-top:-15px;" src="images/feature_importances.png" />

# Let's take a closer look at the distribution of the population sizes for the counties

In [42]:
population_distribution = pd.read_csv('population_summary_df.csv')

In [43]:
population_distribution[''] = population_distribution['Unnamed: 0']

In [44]:
population_distribution.drop('Unnamed: 0', axis = 1, inplace = True)

In [45]:
population_distribution.index = population_distribution['']

In [46]:
population_distribution.drop('', axis = 1, inplace = True)

In [47]:
population_distribution

Population Estimate of All Counties  \
                                            
count                               2,962   
mean                              108,844   
std                               339,283   
min                                 1,183   
25%                                12,949   
50%                                28,180   
75%                                74,528   
max                            10,137,915   

      Population Estimate of Counties with Zero Mortality  \
                                                            
count                                              2,261    
mean                                              30,427    
std                                               37,773    
min                                                1,183    
25%                                               10,242    
50%                                               19,920    
75%                                               37,304    
max                                              849,843    

      Population Estimate of Counties with Greater than Zero Mortality  
                                                                        
count                                                701                
mean                                             361,770                
std                                              631,186                
min                                               18,646                
25%                                               91,251                
50%                                              174,827                
75%                                              390,918                
max                                           10,137,915

### Observations from Distribution of Population from Counties

<ul style="padding-left:40px">
    <li>Minimum population size for counties with zero mortality rate is 18646</li>
</ul>

<ul style="padding-left:40px">
    <li>1078 counties have zero mortality rate and have population less than 18646</li>
</ul>

<ul style="padding-left:40px">
    <li>1183 counties have zero mortality rate and have population greater than or equal to 18646</li>
</ul>

<ul style="padding-left:40px">
    <li>701 counties have greater than zero mortality rate and have population greater than or equal to 18646</li>
</ul>

### Observations from Distribution of Population from Counties

<ul style="padding-left:40px">
    <li>Remove 1078 counties that have zero mortality rate and have population less than 18646</li>
</ul>

<ul style="padding-left:40px">
    <li>Redo prediction analysis with the remaining 1884 counties</li>
</ul>

### Boxplot of Bootstrap RMSE Distributions for All Methods (Population >= 18646)
<img src="images/pop_adj_boxplot_bootstrap_dist.png" />

### Predicted vs Actual Opioid Mortality Rate (Population >= 18646)
<img style="margin-top:-15px;" src="images/pop_adj_predicted_actual_regression_all.png" />

### Predicted vs Actual Opioid Mortality Rate (Population >= 18646)
<img style="margin-top:-15px;" src="images/pop_adj_predicted_actual_regression.png" />